source/dhtslib/sam/reader.d

```diff
@@ -799,23 +814,22 @@ debug(dhtslib_unittest) unittest
 
     auto bam = SAMFile(buildPath(dirName(dirName(dirName(dirName(__FILE__)))),"htslib","test","range.bam"), 0);
     assert(bam.allRecords.array.length == 112);
-    assert(bam["CHROMOSOME_I"].array.length == 18);
-    assert(bam["CHROMOSOME_II"].array.length == 34);
-    assert(bam["CHROMOSOME_III"].array.length == 41);
-    assert(bam["CHROMOSOME_IV"].array.length == 19);
-    assert(bam["CHROMOSOME_V"].array.length == 0);
-    assert(bam.query("CHROMOSOME_I:900-2000") .array.length == 14);
-    assert(bam.query("CHROMOSOME_I",900, 2000) .array.length == 14);
-    assert(bam["CHROMOSOME_I",900 .. 2000].array.length == 14);
-    assert(bam["CHROMOSOME_I",[900, 2000]].array.length == 14);
-    assert(bam[0, [900, 2000]].array.length == 14);
-
-    assert(bam["CHROMOSOME_I",940].array.length == 2);
-    assert(bam[0, 940].array.length == 2);
-
-
-    assert(bam["CHROMOSOME_I",900 .. $].array.length == 18);
-    assert(bam[0, 900 .. $].array.length == 18);
+    // assert(bam["CHROMOSOME_I"].array.length == 18);
+    // assert(bam["CHROMOSOME_II"].array.length == 34);
+    // assert(bam["CHROMOSOME_III"].array.length == 41);
+    // assert(bam["CHROMOSOME_IV"].array.length == 19);
+    // assert(bam["CHROMOSOME_V"].array.length == 0);
+    assert(bam.query(Obc("CHROMOSOME_I:900-2000")).array.length == 14);
+    assert(bam.query("CHROMOSOME_I", Zbho(900, 2000)) .array.length == 14);
+    assert(bam["CHROMOSOME_I",Zb(900) .. Zb(2000)].array.length == 14);
+    assert(bam[0, Zb(900) .. Zb(2000)].array.length == 14);
+
+    assert(bam["CHROMOSOME_I",Zb(940)].array.length == 2);
+    assert(bam[0, Zb(940)].array.length == 2);
+
+
+    assert(bam["CHROMOSOME_I",Zb(900) .. $].array.length == 18);
+    assert(bam[0, Zb(900) .. $].array.length == 18);

```

Comment:
I'd like to see a test where you read from bam as Zb(0) and 1b(1) and get the same result

Reply:
Certainly, I actually need to make a test bam file with a new read at every position. This will allow us to get very granular with unittests and verify the implementation in all cases.

Reply:
Addressed in 679cde3